src/test/scala/plugin/inline/InlineErrorCheckerTest.scala

```diff
@@ -122,11 +124,53 @@ class InlineErrorCheckerTest extends FunSuite with InlineErrorChecker with Inlin
         PredicateAccessPredicate(PredicateAccess(Seq(), calledPredId)(), FullPerm.apply()())(),
       )()
     )
+    val predicateForF = Predicate(calledPredId, Seq(), None)()
+    val program = programCopy(predicates = Seq(predicateForF))
     val pred = Predicate("loop", formalArgs = Seq(), body = maybeRecursiveBody)()
 
-    val maybeResult = nonRecursivePredsCalledBy(pred)
+    val maybeResult = nonRecursivePredsCalledBy(pred, program)
     maybeResult.fold(false) { result =>
       result.size == 1 && result(calledPredId)
     }
   }
+
+  test("predicatesCalledBy should traverse subpredicates") {
```

Comment:
_testing is my passion_ (jk kill me now lol)

Reply:
I'm realizing I wrote .vpr file tests but not tests in this file :speak_no_evil: